### ISSUE TITLE
Packer send wrong ssh private key to Ansible

### ISSUE THREAD

**Issue by Sabbasth** [hashicorp/packer#5155]
- Host is Debian Jessie

```bash
$packer version
Packer v1.0.3
```

```yaml
{
    "description":"Build AMI for Jenkins slaves",
    "min_packer_version":"0.10.1",

    "variables": {
        "aws_access_key": "{{env `AWS_ACCESS_KEY`}}",
        "aws_region": "{{env `AWS_REGION`}}",
        "aws_secret_key": "{{env `AWS_SECRET_ACCESS_KEY`}}",
        "aws_source_ami": "ami-402f1a33",
        "aws_ssh_username": "admin",
        "aws_subnet_id": "subnet-3cc3155b",
        "aws_volume_size":"100",
        "aws_vpc_id": "vpc-87e839e3",
        "csq_app": null,
        "csq_app_version": null,
        "packer_playbook": null
    },

    "builders": [
        {
            "type": "amazon-ebs",
            "access_key": "{{ user `aws_access_key` }}",
            "secret_key": "{{ user `aws_secret_key` }}",
            "region": "{{ user `aws_region` }}",
            "source_ami": "{{ user `aws_source_ami` }}",
            "instance_type": "t2.small",
            "vpc_id":"{{ user `aws_vpc_id` }}",
            "subnet_id":"{{ user `aws_subnet_id` }}",
            "ssh_username": "{{ user `aws_ssh_username` }}",
            "ami_name": "jenkins-slave-{{ user `csq_app` }}-{{ user `csq_app_version` }}",
            "associate_public_ip_address": false,
            "ami_block_device_mappings": [
                {
                    "device_name": "/dev/xvda",
                    "volume_size": "{{ user `aws_volume_size` }}",
                    "delete_on_termination": true
                }
            ],
            "launch_block_device_mappings": [
                {
                    "device_name": "/dev/xvda",
                    "volume_size": "{{ user `aws_volume_size` }}",
                    "delete_on_termination": true
                }
            ]
        }
    ],

    "provisioners": [
        {
            "type": "ansible",
            "playbook_file": "{{ user `packer_playbook` }}",
            "user": "{{ user `aws_ssh_username` }}"
        }
    ]
}
```
See which private key is used by Ansible
```bash
ps aux | grep IdentityFile
bastien.semene    1940   0,0  0,0  2433828   1684 s002  R+   10:26     0:00.00 grep --color=auto --exclude-dir=.bzr --exclude-dir=CVS --exclude-dir=.git --exclude-dir=.hg --exclude-dir=.svn IdentityFile
bastien.semene    1583   0,0  0,0  2454344   7472 s000  S+   10:25     0:00.02 ssh -C -q -o ControlMaster=auto -o ControlPersist=60s -o StrictHostKeyChecking=no -o Port=51083 -o IdentityFile="/var/folders/ll/by_nmkg14ng_kvyn_wty1dlc0000gn/T/ansible-key796599073" -o KbdInteractiveAuthentication=no -o PreferredAuthentications=gssapi-with-mic,gssapi-keyex,hostbased,publickey -o PasswordAuthentication=no -o User=admin -o ConnectTimeout=5 -o ControlPath=/Users/bastien.semene/.ansible/cp/ansible-ssh-%h-%p-%r 127.0.0.1 /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1500625558.7-27365017447877 `" && echo ansible-tmp-1500625558.7-27365017447877="` echo $HOME/.ansible/tmp/ansible-tmp-1500625558.7-27365017447877 `" ) && sleep 0'
```
diff between private key file provided by `-debug` and the one provided to ansible:
```bash
diff ec2_amazon-ebs.pem /var/folders/ll/by_nmkg14ng_kvyn_wty1dlc0000gn/T/ansible-key796599073
2,23c2,27
< MIIEowIBAAKCAQEAhCCEEkDEraXSzYmqKigk1zrmgo31KFNoGpvj1YOOtr4hwIkpe+SnvTp4csuA
< 67e6XwmlkQ3/4N6nL+kGWOFsN4jFQWGxSY/W8OgVUi5U6Q5vX6PjLbr05P7HzRdMmUEukx9BHJPe
< hlr5lMWAFweZ/5DgUaeFU0vbD7uk1lt21YzdTx5euPeGy3+t5bry8sXSSWT0eNy8Swvur91QFh2+
< Jql0dZpo73oNJwP5yq+QRtG+nuPZP9CwWJ98bamDUfQDPcQ+Nw5sMUY6wcppeomdaDkl7LJixysb
< y2V18TYH9z5lyc+9xYYXB3ubqxOPUMiGcVCWr/32V2JL1RcTjePdpQIDAQABAoIBAGTVZ9WCRlb7
< aSBhAlM/5bW4Brj123XWCWndNIf66Yw/GAUW6GjbvMOl/swk0JOT2JC1DPzhgTi80zy2CpkEGBFd
< uKRICQPJ2VBq3fKk1zoxQP/tU+/PcRvd8N9s+5Et9Hxk6s0DcFx6y2QlQbbRn04x42+NeCt0hY7B
< NnA8xzj7OiY7NIPSOwarXZRpQYigHx9W/qBZnOhjmSU080xzh+K8XaRWBB4gCzQISc/XCn7T2JTk
< cr3dbiphCtn9+BgDD6bLf3hXxR5Bdmnicu9axSRdQrVrutBxED/edpfPSUa//UdQOMOUPGvQaoPG
< OwveaIjUNAzFG4B3qGxmhuJEwpUCgYEAuMoOn3XUu2oDn2+sd90eW23MSy+Dfsw/8ITBy9EGhSpB
< zuAHQ0+h1LN74bclPmUXl9D3gs6uTeUAe4cCNDk+su8uS4Zc3vs7+DOPq/yrf+InvP65vRBJ5gF4
< 0YjQo9jp6Ndog7y9T7Bn/dtaG+f1ZCpNwSB/s0f1WcGsR/OBACMCgYEAtwswZiuNJD4zdcStUTil
< T4pe7idYbNDENxTjQx1pyrzspJ+fxEvLIYPbZIpYgqaxtDqZKMyJ+h4q0sx+dpNI0b+hodG8tR6g
< EFOJZtT69JCn8tvEjOYwuM/CypHV9LcEgG5W06VEF7xQyRjo1qafbNOcjCWCbU9rwOmml8eYI5cC
< gYAkUcp3+v/Z6idWIevjDznVPIpnyKnutRRoIJ+6gJXofHq8v3mN+EvVc4WIwqd2UQZHF5JeSmMl
< +jmTDeEYBJxXLP2LaJg+V2DdEr1NT3chxoCK/AHjsBzmMSne/SYM3ihzVEs9gU4wKHr4ATH/xCKO
< EmZ/6/RNd3o4rwE+jxSuwwKBgCbwlG1H2pDZQq1NF+nteQ2AzE0xxZfwma12HhrLCRR+OIGVC7IA
< xcfteyDaZZX1zFwhEzAgw0GYZVfO9SBYmOc2MzZvnguR5k9S1iowKOsaUwBRTHEFD1xhpnjdqXyT
< 6rs/vezsfZ3nC/CcAfyYGdZW9jjCSgrjGHNDwZMMfQslAoGBAK8X9bxxzwr8MA0RWMJsQHwTHFVF
< sVVGH5vUJ2SZPBC7gOGWkzvo+v8C9zuiOtiOFYE7UHdTevaDHJJ1OVPcITt+bpWbYH17BZLMhPzs
< e2OgnEO1o9gmllbwSZNYW+NcuMsG6HhUj1UpRmgx9rPhf4egJ/vCQaBGGwDL7aGBt6S0
< -----END RSA PRIVATE KEY-----
\ No newline at end of file
---
> MIIEowIBAAKCAQEAxjIeOo8Y/rYmgHWaA94ImHYimchqJQIMN0Egtw50teBy4U5b
> 87XYnqZUHXrOcSy8IkbvHWhmz0Itto4U72bzS4UcL7zV7JibsdPHA7SkNBZV7zta
> +q+CPnAmJmztBw12UtBrezW3H0YDGOgvav/m4P138xXhexYEFi5RTjPDISQGSUvG
> g6kn4XZLhvJFczWhvz567B/ofTtHHPhxcfkwdJcgKBMMNCvEiT+kc8WyN5mxorYQ
> IaPWWwEkSr62L0wnrOnL7IylwSeTsQ4Dragut0BagkBHC89hVxi5YUBZ5KoW48wp
> NNjjCXQf5tHO/9yr+YRIRC4J9z5ZwvEMkPFACQIDAQABAoIBAByndE1dEO0KYqih
> 2AGFIVMBiWEYY5g0IpX+eX9dvKvq5lECpjKm8MmHHJQihCRCpnSCMuHajTqbw8Yn
> 5ojyJmydswSH1ZneOmxMRPKZNEA/lm7+8tlZdIgYaLuXdfFWFexFK7TlgDMqSRpZ
> vzlGBBg1LRlc6XuFJFQcoC6hNy3cKVSiXF3eancPHzpNtaQi5VNsVqVSx2snzZTy
> aF0RoVgzkNXPjU6bvjsZNxGNE1FwiJO7Yf4B9FAnO3yt+6w41tXJYgET7DqxhJPO
> vOoOMZugBaPHLwEZvfOkqSUqXfneUZ1pRWoVoL+lz/Tpr9BiTRHS6qgxz1DzOQ3k
> Mszmzr0CgYEA5MSeaZghJ91A4yVyG1Z4FZ6UePTpggGaIeklE2FLF8Yv+3fTP+WG
> 67PsUeUvpBl4TUYTtaPd296n+qYn7Bph3Vv90pAw351SFCa7wYoa8tvvoqqxawkA
> CJJZ6CPfTPx65ldbfrL5QzkdDV4cYvAsHKHehtfPj9htIyI3f+V6hicCgYEA3cna
> UlmAloYEOfOeQ7rdcZY7A5GjexmL443nz+GPXYB5U2SrB/yWZX7oin3Ur/+ag/B0
> DSB64q33a2iDppnVDV9e6pG40TpY4bd17iTfZ/SrKzdezKgkj8SKKgAIiLnrtiKc
> oTr51jHOv3WrmEL3tt306Bxa0NRCO0WcmBGhlk8CgYAzCX6B9lOnTiA0zNzZJoBh
> 6mAbYCBcRbk6SYZg/XrAUhHQW1pAzhc8rIZTevsW9Cu3qaOqCWXssI7vR3RuozAk
> L2rAODT/Hlz6a+3Fbuc2OUUMGGK+i2J7hxMRzDb8OPO+Qc0CxjEAA1Ts74sLs6NT
> u/UmRi1jXRh0FyUknnCoTwKBgQCNjnkXGGMXL6Bg8Jnx9sHB7rXp2Dm0pyB+86Ki
> LwAXGJnXXShAZ7oCO9bmFdv7HTQcxAr+Gd308j45wrJMUmEGMPne03VXrVrvbe4K
> OmSQ8DuT0Lpj+2mijTn2BqFryuakEEFSMtclHBMuZkdIXm3GvLHEt9xK7z76n7qL
> 4Kz2xwKBgH5K+KdAHBrTQK2RVXwucijQW+tQ0YNZKFuI/9nCqrI9WIEZz3NtVRzb
> 3xNJAaLjlkSpnnkHmYRQepIqKdphMjbD0LLIRHpPCEUw8CKG8kp7KFKL4E9zwmAK
> Lmlg7fDgoeyx1GVjN4N4XrzVJ2gao5rOVfLrEH4Yl0t5WnZe70ls
> -----END RSA PRIVATE KEY-----
```
Link to the output log of Packer: https://gist.github.com/Sabbasth/8229e7976b3b96059030a412cb39492a#file-packer_output-log

manual ssh connections:
```bash
$ ssh 10.1.36.238 -l admin -i ec2_amazon-ebs.pem
The authenticity of host '10.1.36.238 (10.1.36.238)' can't be established.
ECDSA key fingerprint is SHA256:oFYbLEn0rWAj/dW+gLlV/BM6dan3SsJ4MAaEqnH/wZE.
Are you sure you want to continue connecting (yes/no)? yes
Warning: Permanently added '10.1.36.238' (ECDSA) to the list of known hosts.

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
admin@ip-10-1-36-122:~$ exit
logout
Connection to 10.1.36.238 closed.
```
```bash
$ ssh 10.1.36.238 -l admin -i /var/folders/ll/by_nmkg14ng_kvyn_wty1dlc0000gn/T/ansible-key796599073
Permission denied (publickey).
```

**Comment by rickard-von-essen** [comment]
Thank you for taking reporting this. This is most likely a duplicate of #5150, which is fixed on master.

Please test if you it works with 1.0.2 or a build from master. If it does this is for sure a dupe. If it doesn't work, please tell and I'll reopen this.

(The analysis that there are different keys used is correct, but intended behavior.)

**Comment by Sabbasth** [comment]
I confirm, fixed on `1.0.2`.

Why do you use different keys ?

**Comment by rickard-von-essen** [comment]
It's because ansible connects to a local proxy (127.0.0.1:51083 in your case) and a temporary keypair is created for connecting towards that.
